app/utils/defaultSettings.js

```diff
@@ -20,7 +20,7 @@ module.exports = {
   microbreakPostponableDurationPercent: 30,
   breakPostponesLimit: 1,
   breakPostponableDurationPercent: 30,
-  mainColor: '#478484',
+  themeName: 'greenClouds',
```

Comment:
Please let's not change the names and values of settings. That's not something that I did even between version 0 and 1. I don't want users to loose settings... (And doing some more code to take care of that would be bad idea. )

Reply:
I don't get how you want to do it then:(

So/does? :
1. the mainColor var should stay and should still be used for setting background color?
2. new class should be at top of HTML to be able to theme it, but does it mean it should be theming everything except background color? 
3. if not using data-* attr on input for font color then I don't know how to do it

Could you clarify for me the solution you've on your mind? .

Reply:
Yes, we need `mainColor` as it also allows people so set whatever color they want for background. 

Currently, we have combination of colors in break windows that kind of works for every theme. (Not perfect, but it works.). With your new theme, we could keep most of the colors, but as I mentioned previously, some things were that good  on white (icon and yellow mouseover). So that's the minimum we need to update.  

Considering that you said you use this on e-ink, I thought we could make that new theme be only black and white colors. But we could do that later, or never :) I guess you can test and see what is enough. 

So I think easiest is to: 
- add class on top of break window (could be even the code of color, doesn't really matter)
- don't change anything for current themes
- when we have  your new theme, update CSS so that we have better colors for that tooltip and colors 

For example let's say your new theme is translated to new `123456` class on breaks:

```
.breaks > :nth-child(2) > a:hover {
  color: #FFFF00;
}

.123456 .breaks > :nth-child(2) > a:hover {
  color: #AAAA00;
}
```

This results in: 
- no change in how old themes works
- no change if users sets custom color for theme
- your new theme had different color on that link hover

So old CSS is not changed, you only kinda "duplicate" CSS that you want to "override" to make your theme better. It being font color, shadow, or anything that is set by CSS 

In future, we could then improve other themes in the same way and find better suiting colors for accessibility or so. 

You can move your "overrides" to specific file if you'd like. But I would not add it dynamically with JS, I would load it right away with the rest of CSS - it's not so huge file to worry about speed or so...

Does this makes sense?

Reply:
1. `when we have your new theme, update CSS so that we have better colors for that tooltip and colors`
do you mean "when we have your new theme **selected** (..)" ?

2. What you mean by `update CSS`? write js code -> if e-ink theme then break window set style ? add class?

3. `For example let's say your new theme is translated to new 123456 class on breaks:` how should I translate it? is it the solution from point 2?

Reply:
4. I'm wondering if you'd agree to reuse the current `mainColor` var + create new one e.g. fontColor (default val=white)  (selecting each theme will update those). 
We could reuse some coding done here and either:
- directly each time theme is selected in preferences change mainColor and fontColor variable,
- in UI create "custom theme" option at the bottom of all other themes and new inputs (in future to be replaced with color picker UI) for mainColor and font color (requires new var - customThemeEnabled(default=true, so prev users will keep their settings)) 
This way we should flexible create clean solution (without hacks in code that are often hard to maintain) and backwards compatible (as current configuration will work after those changes).

Reply:
OK, so let me try to explain differently what I mean. 

Some facts: 
- The new feature here is adding new Theme (Heaven White). 
- Theme means how Break Window is styled. 
- Old themes share same CSS (except background color). 
- CSS is across different files (we import `normalize.css`, `commons.css` and `break.css`)
- we choose theme to be used in Preferences window and save it as `#` value (this is what I meant by `selected`)

Adding new theme should be as easy as adding one more option in Preferences, but it is not in this case. 

Problem: Heaven White needs different colors for specific elements on Break Window. 

Solution: Leave current CSS as is, but add code for Heaven White  that will take the priority over current CSS in Break Window when Heaven White is current theme, (You can do that by using CSS class - (https://www.w3schools.com/css/css_specificity.asp) ). 

You are already adding new CSS class do achieve that here: 

```
document.body.classList.add(theme)
``` 
(We need to change it to use the old settings as I mentioned but that's detail: you can take `mainColor` from settings, remove `#` and set that as class to be added.)
 
With that change, we now have additional class on Break Window telling us what theme are we using. 

So for example font color is currently in `break.css`:

```
.breaks {
...
  color: #FFFFFF;
...
}
```
If we added `FFFFFF` class for Heaven White, in CSS you just do (or similar depending where you put the class): 

```
.FFFFFF .breaks {
...
  color: #000000;
...
}

```
This should override the `.breaks` selector. right? 

It's pretty much the same what you did in your new file `themes.css` but there is no need to specify all previous ones, you can just do override for new theme. As example: when user did not change theme in Preferences, `478484` class would be added to Break window. But that does not change the old behaviour, so  all good. 

Therefore there is no need to save font-color in preferences. The same way you can override any other theming changes needed. 

This also allows for later gradual changes for current themes if there's need to make them better.  

We could of course add `fontColor` value to settings to allow more configuration for users, but that's for future. I would like to keep PRs simple, so only one feature at time. Also, adding font color would not be enough, though, as we also have the need for mouse-over color etc.. I a not sure how theming will work in future...

In general, I would prefer to keep config simple, too much ogf a choice makes bad UX.